### PR TITLE
[SEMI-MODULAR] Makes security record photo generation not happen all at once

### PR DESCRIPTION
**Pull request by monster860**
## About The Pull Request
Makes security record photo generation happen in a way as to *not* produce a massive amount of lag. In addition, potentially maybe fix a race condition that causes glitchy photo generation.

I'm not 100% sure if this is the best solution to the problem, it might be a further improvement to make a subsystem for generating flat icons.

### CODE REVIEW

**Comment by thestubborn** [comment]
non modular

changelog

**Comment by nevimer** [comment]
https://github.com/Skyrat-SS13/Skyrat-tg/pull/10346

This does this, it's calling the wrong proc for generating sprites.

**Review by Gandalf2k15 — APPROVED** [review]
Looks good

File: modular_skyrat/modules/deferred_icon/code/deferred_icon.dm [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+	if(job)
+		body.dna.species.pre_equip_species_outfit(job, body, TRUE)
+	if(outfit)
+		body.equip_outfit_and_loadout(outfit, prefs, TRUE)
+
+	var/list/out_list = list()
+	COMPILE_OVERLAYS(body)
+	for(var/D in showDirs)
+		body.setDir(D)
+		out_list["[D]"] = image(body.appearance)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/deferred_icon/code/deferred_icon.dm:22]: Please avoid single-letter vars.

File: modular_skyrat/modules/deferred_icon/code/deferred_icon.dm [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+#define DUMMY_HUMAN_SLOT_MANIFEST "dummy_manifest_generation"
+/datum/datacore/proc/get_id_image(mob/living/carbon/human/H, client/C, show_directions = list(SOUTH))
+	var/datum/job/J = H.mind.assigned_role
+	var/datum/preferences/P
+	if(!C)
+		C = H.client
+	if(C)
+		P = C.prefs
+	return get_human_image(null, J, P, DUMMY_HUMAN_SLOT_MANIFEST, show_directions)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/deferred_icon/code/deferred_icon.dm:217]: Here too, single-letter var bad.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Pull request closed without merging**